security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -567,11 +577,68 @@ func (sc *SecretCache) generateGatewaySecret(token string, connKey ConnKey, t ti
 	}, nil
 }
 
+// If there is an existing secret under a well known path, return true.
+// Otherwise, return false.
+func (sc *SecretCache) secretExists() bool {
+	b, err := ioutil.ReadFile(existingCertChainFile)
```

Comment:
I would break the 'root cert' and 'key/cert' in separate methods. 
There are 2 different cases in the SDS call anyways.

The reason is that we may still keep the root cert in the Secret, without the key/cert - for cases where signing with
K8S is not viable.

Reply:
Done.